### PR TITLE
Playout without self-atari of large chains

### PR DESCRIPTION
**Pull request by iopq**
This runs the playouts without simulating blunders by the player or the opponent.

That said, it is still allowed in the game tree as a ko threat. So you'll still see these moves being played either as a threat or as desperation move (we can actually resign if our opponent has a 100% win refutation though but we don't check this)

### CODE REVIEW

**Comment by iopq** [comment]
It beat the light playout version 7-3 in a ten game match

File: src/playout/mod.rs [hunk 1]
```diff
@@ -20,61 +20,45 @@
...
-        Some(s) => {
-            match s.as_ref() {
-                "no-eyes-with-pass" => Box::new(NoEyesWithPassPlayout::new()),
-                "simple" => Box::new(SimplePlayout::new()),
-                "simple-with-pass" => Box::new(SimpleWithPassPlayout::new()),
-                _ => Box::new(NoEyesPlayout::new()),
-            }
-        },
-        None => Box::new(NoEyesPlayout::new())
+    match opt.as_ref().map(::std::ops::Deref::deref) {
```
Comment (ujh) [inline src/playout/mod.rs]: Could you please add a test for the `"no-self-atari"` case?

File: src/engine/uct/mod.rs [hunk 2]
```diff
@@ -112,15 +112,13 @@ fn spin_up_worker<'a>(config: Arc<Config>, board: Board, color: Color, send_to_m
...
-                    // TODO: We do a clone here and then one when we
-                    // call run(). One is enough!
                     let mut b = board.clone();
                     for &m in moves.iter() {
                         b.play_legal_move(m);
                     }
                     // Playout is smart enough to correctly handle the
                     // case where the game is already over.
-                    let playout_result = config.playout.run(&b, None, &mut rng);
+                    let playout_result = config.playout.run(&mut b, None, &mut rng);
```
Comment (ujh) [inline src/engine/uct/mod.rs]: :+1:

File: src/board/mod.rs [hunk 3]
```diff
@@ -563,6 +563,41 @@ impl Board {
         self.board[c.to_index(self.size)].chain_id = -1;
         self.board[c.to_index(self.size)].color = Empty;
     }
+    
+    pub fn liberty_count(&self, c: Coord) -> usize {
```
Comment (ujh) [inline src/board/mod.rs]: Could you please add a few tests for these functions?

Comment (iopq) [inline src/board/mod.rs]: I added tests and now I do non-overlapping liberties which is slower, but more correct

**Comment by ujh** [comment]
Awesome. I have a bit of a problem with my big toe and the rest of the family had a cold so we're all out. Feel free to merge it and/or make it work with the latest nightly build.

**Pull request merged**